### ISSUE TITLE
make Tensor.backward return self

### ISSUE THREAD

**Issue by Yura52** [pytorch/pytorch#40159]
## 🚀 Feature
Make the method `torch.Tensor.backward()` return the tensor itself.

## Motivation
The suggested feature enables method chaining which is especially useful in the following pattern:
```python
loss = loss_fn(...)
loss.backward()
loss = loss.item()
```
With the suggested change it becomes more concise:
```python
loss = loss_fn(...).backward().item()
```

**Comment by ailzhang** [comment]
Hmmmm return `loss` is essentially the input `loss.backward()` function is doing(which is the backprop) . Personally I don't see a strong motivation of adding it as output.

**Comment by Yura52** [comment]
Well, after more thinking I rather agree that returning `self` from `backward` is hard to justify from the perspective of semantics. The particular pattern I described above should be handled with a separate method or function. For example:
```python
class Tensor:
    def backward_item(self):
        self.backward()
        return self.item()
```
The motivation is the same:
```python
loss = loss_fn(...)
loss.backward()
loss = loss.item()
# vs
loss = loss_fn(...).backward_item()
```
If such functionality is out of scope for PyTorch, I think the issue can be closed.
